portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -0,0 +1,46 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.String;
+
+
+@WebServlet("/chart-data")
+public class ChartServlet extends HttpServlet {
+
+  private Map<String, Integer> popularCommenters = new HashMap<>();
```

Comment:
You can add "final" to make this popularCommenters map immutable. This means that the pointer to this map is immutable, but you can still modify this map by clearing it, adding elements to it or removing elements from it.

private final Map<String, Integer> popularComments = new HashMap<>();

Reply:
Thank you for the recommendation! I was just thinking what's the benefit of making it final verus if it was not final?